x/mongo/driver/operation.go

```diff
@@ -781,37 +802,38 @@ func (op Operation) retryable(desc description.Server) bool {
 
 // roundTrip writes a wiremessage to the connection and then reads a wiremessage. The wm parameter
 // is reused when reading the wiremessage.
-func (op Operation) roundTrip(ctx context.Context, conn Connection, wm []byte) ([]byte, error) {
-	err := conn.WriteWireMessage(ctx, wm)
+func (op Operation) roundTrip(ctx context.Context, conn Connection, wm *[]byte) ([]byte, error) {
+	err := conn.WriteWireMessage(ctx, *wm)
 	if err != nil {
 		return nil, op.networkError(err)
 	}
-
 	return op.readWireMessage(ctx, conn, wm)
 }
 
-func (op Operation) readWireMessage(ctx context.Context, conn Connection, wm []byte) ([]byte, error) {
+func (op Operation) readWireMessage(ctx context.Context, conn Connection, wm *[]byte) ([]byte, error) {
 	var err error
 
-	wm, err = conn.ReadWireMessage(ctx, wm[:0])
+	*wm, err = conn.ReadWireMessage(ctx, (*wm)[:0])
 	if err != nil {
 		return nil, op.networkError(err)
 	}
 
 	// If we're using a streamable connection, we set its streaming state based on the moreToCome flag in the server
 	// response.
 	if streamer, ok := conn.(StreamerConnection); ok {
-		streamer.SetStreaming(wiremessage.IsMsgMoreToCome(wm))
+		streamer.SetStreaming(wiremessage.IsMsgMoreToCome(*wm))
 	}
 
 	// decompress wiremessage
-	wm, err = op.decompressWireMessage(wm)
+	*wm, err = op.decompressWireMessage(*wm)
 	if err != nil {
 		return nil, err
 	}
 
 	// decode
-	res, err := op.decodeResult(wm)
+	b, err := op.decodeResult(*wm)
+	res := make([]byte, len(b))
+	copy(res, b)
```

Comment:
Can you add a comment explaining why the copy of `b` is necessary?
```suggestion
        // Copy b to extend the lifetime. b may be a subslice of wm. wm will be added back to the memory pool and reused.
	copy(res, b)
```